### ISSUE TITLE
Segmenting only from endpoint ROIs

### ISSUE THREAD

**Issue by arokem** [yeatmanlab/pyAFQ#918]
We have use-cases where we'd like to segment tracts based on cortical endpoints only. For example, functional ROIs derived from fMRI. In principle, this should work by defining a bundle dict like this: 

```
bundles = abd.BundleDict({
 "L_MTPT": {
 "start": ImageFile(
 suffix="mask";
 filters={'label': "lROI1",
 'scope': 'roi'}
 ),
 "end": ImageFile(
 suffix="mask";
 filters={'label': "lROI2",
 'scope': 'roi'}
 ),
 "cross_midline": False,
 "space": "subject"},
 "R_MTPT": {
 "start": ImageFile(
 suffix="mask";
 filters={'label': "rROI1",
 'scope': 'roi'}
 ),
 "end": ImageFile(
 suffix="mask";
 filters={'label': "rROI2",
 'scope': 'roi'}
 ),
 "cross_midline": False,
 "space": "subject"
 }
})
```

Where `roi` is the name of a pipeline used to generate the ROIs. However, in practice this doesn't work.  The first reason we run into off the bat is probably prosaic and solveable, and has to do with the fact that the ROIs that gets passed here: https://github.com/yeatmanlab/pyAFQ/blob/101db5bfb5aa6722b783e9da59dd38d25046687e/AFQ/segmentation.py#L1211 do not conform with the assumptions of cdist and the code crashes here: https://github.com/yeatmanlab/pyAFQ/blob/101db5bfb5aa6722b783e9da59dd38d25046687e/AFQ/segmentation.py#L1184. 

But even if we solve this issue, I think that the more difficult problem will arise in this line of code: https://github.com/yeatmanlab/pyAFQ/blob/101db5bfb5aa6722b783e9da59dd38d25046687e/AFQ/segmentation.py#L772, because we use the distance to the waypoint ROIs to determine the orientation of the streamlines. I think that means that we need to think of an alternative scheme where streamlines are checked for their distance to the endpoint ROIs and then oriented based on that, before going into the `clean_by_endpoints` function. But not sure how deep of a change we want to make here.

**Comment by arokem** [comment]
Maybe we can refactor clean_by_endpoints so that under some conditions it orients the streamlines based on proximity to the endpoints. Might require some significant changes to the logic of that function but then at least it will maybe be limited just to that function.

**Comment by 36000** [comment]
If this is due to the L_MTPT error, I *think* that might be because the ROI is in a different space than the DWI. So I would double check the Nifti header. If they are in the same space, then I may have a bug where I am resampling subject space ROIs when I shouldn't be. If they aren't in the same space, maybe I should auto-resample them.

**Comment by arokem** [comment]
OK - I see where this is implemented: https://github.com/yeatmanlab/pyAFQ/blob/master/AFQ/segmentation.py#L474-L477. What happens if the streamlines are pretty far away from the endpoints? The tolerance for endpoint filtering is usually larger than tolerance for waypoint selection. Should we change the tolerance for waypoint selection so that they get included in the first place? I will also check the ROI, but I'm pretty sure everything is in the T1w space.

**Comment by 36000** [comment]
Found possible problem #919

**Comment by arokem** [comment]
Excellent! I will give that a try sometime tonight and let you know if that makes the difference.

**Comment by arokem** [comment]
Closed by #919
